app/controllers/posts_controller.rb

```diff
@@ -27,4 +27,11 @@ def create
       end
     end
   end
+
+  def destroy
+    @post = Post.find(params[:id])
+    @post.comments.destroy_all
+    @post.destroy
+    redirect_to user_path(params[:user_id])
+  end
```

Comment:
- [ ] Kindly use `load_and_authorize_resource` at the start of your post controller. This will check if the currently logged-in user is authorized to do a certain action. You can read more about this [here](https://github.com/CanCanCommunity/cancancan#:~:text=Setting%20this%20for%20every%20action%20can%20be%20tedious%2C%20therefore%20the%20load_and_authorize_resource%20method%20is%20provided%20to%20automatically%20authorize%20all%20actions%20in%20a%20RESTful%20style%20resource%20controller.%20It%20will%20use%20a%20before%20action%20to%20load%20the%20resource%20into%20an%20instance%20variable%20and%20authorize%20it%20for%20every%20action.)

Reply:
Thanks for the review and link to more information